integrations/aws/aws/utils.py

```diff
@@ -0,0 +1,297 @@
+    is_global = is_global_resource(kind)
+    async for session in get_sessions(None, None, is_global):
+        region = session.region_name
+        logger.info(f"Resyncing {kind} in region {region}")
+        account_id = await _session_manager.find_account_id_by_session(session)
+        next_token = None
+        while True:
+            try:
+                async with session.client("cloudcontrol") as cloudcontrol:
+                    params = {
+                        "TypeName": kind,
+                    }
+                    if next_token:
+                        params["NextToken"] = next_token
+
+                    response = await cloudcontrol.list_resources(**params)
+                    next_token = response.get("NextToken")
+                    resources = response.get("ResourceDescriptions", [])
+                    if not resources:
+                        break
+                    page_resources = []
+                    for instance in resources:
+                        described = {
+                            "Identifier": instance.get("Identifier"),
+                            **json.loads(instance.get("Properties", {})),
+                        }
+                        described.update(
+                            {
+                                KIND_PROPERTY: kind,
+                                ACCOUNT_ID_PROPERTY: account_id,
+                                REGION_PROPERTY: region,
+                            }
+                        )
+                        page_resources.append(
+                            fix_unserializable_date_properties(described)
+                        )
+                    yield page_resources
+            except Exception:
+                logger.exception(
+                    f"Failed to list CloudControl Instance in account {account_id} kind {kind} region: {region}"
+                )
+                break
+            if not next_token:
+                break
+
+
+def get_matching_kinds_from_config(kind: str) -> list[ResourceConfig]:
+    return list(
+        filter(
+            lambda resource_config: resource_config.kind == kind
+            or (
+                isinstance(resource_config, AWSResourceConfig)
+                and kind in resource_config.selector.resource_kinds
+            ),
+            typing.cast(AWSPortAppConfig, event.port_app_config).resources,
+        )
+    )
+
+
+def get_resource_kinds_from_config(kind: str) -> list[str]:
+    """
+    Gets the `resourceKinds` property from the port_app_config that match the given resource kind
+    """
+    resource_config = typing.cast(AWSResourceConfig, event.resource_config)
+    resource_config
+    if (
+        resource_config.kind == kind
+        and hasattr(resource_config.selector, "resource_kinds")
+        and resource_config.selector.resource_kinds
+    ):
+        return resource_config.selector.resource_kinds
+    return []
```

Comment:
why is that function needed? u r using it only for the cloud resource and this kind always has resource kinds property

Reply:
Yes but the user can always edit this `resourceKinds`

Reply:
I think splitting to functions will make the code easier to read, WDYT?